File: frontend/lib/model/schedule_day.dart
import 'package:poligon/model/path.dart';
import 'package:poligon/model/schedule_element.dart';

/// Author: Łukasz Piętka (FUT 2025)
class ScheduleDay {
  final int id;
  final DateTime day;
  final TrainingPath path;
  final List<ScheduleElement> elements;
  final DateTime? lastModified;

  ScheduleDay({required this.id, required this.day, this.path = TrainingPath.wspolpraca, required this.elements, this.lastModified});
}

File: backend/src/main/java/me/havooc/poligon/model/ScheduleDay.java
/**
 * @Author Łukasz Piętka (FUT 2025)
 */
package me.havooc.poligon.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ScheduleDay {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Enumerated(EnumType.ORDINAL)
    private TrainingPath schedulePath;
    private LocalDate day;
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    private List<ScheduleElement> elements;
    @UpdateTimestamp
    private LocalDateTime lastModified;
}